# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBIndex.cpp
# Commit: 300dc3034eeb
# Full Hash: 300dc3034eebd2fc8bcf24e47d886b27b793703e
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-10 15:51:07
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Reduce code duplication for different open cursor variants. r=ttung
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40956
# ==============================================================================

diff -r d6aa1c5bc7d9 -r 300dc3034eeb dom/indexedDB/IDBIndex.cpp
--- a/dom/indexedDB/IDBIndex.cpp	Tue Sep 10 16:47:26 2019 +0300
+++ b/dom/indexedDB/IDBIndex.cpp	Tue Sep 10 16:47:41 2019 +0300
@@ -455,28 +455,12 @@
 
   IDBCursor::Direction direction = IDBCursor::ConvertDirection(aDirection);
 
-  OpenCursorParams params;
-  if (aKeysOnly) {
-    IndexOpenKeyCursorParams openParams;
-    openParams.commonIndexParams().commonParams().objectStoreId() =
-        objectStoreId;
-    openParams.commonIndexParams().indexId() = indexId;
-    openParams.commonIndexParams().commonParams().optionalKeyRange() =
-        std::move(optionalKeyRange);
-    openParams.commonIndexParams().commonParams().direction() = direction;
+  const CommonIndexOpenCursorParams commonIndexParams = {
+      {objectStoreId, std::move(optionalKeyRange), direction}, indexId};
 
-    params = std::move(openParams);
-  } else {
-    IndexOpenCursorParams openParams;
-    openParams.commonIndexParams().commonParams().objectStoreId() =
-        objectStoreId;
-    openParams.commonIndexParams().indexId() = indexId;
-    openParams.commonIndexParams().commonParams().optionalKeyRange() =
-        std::move(optionalKeyRange);
-    openParams.commonIndexParams().commonParams().direction() = direction;
-
-    params = std::move(openParams);
-  }
+  const auto params =
+      aKeysOnly ? OpenCursorParams{IndexOpenKeyCursorParams{commonIndexParams}}
+                : OpenCursorParams{IndexOpenCursorParams{commonIndexParams}};
 
   RefPtr<IDBRequest> request = GenerateRequest(aCx, this);
   MOZ_ASSERT(request);